sceptre_cdk_handler/cdk.py

```diff
@@ -0,0 +1,225 @@
+        """The synthesizer args for CDK Bootstrapless Synthesizer. Used only with the "bootstrapless"
+        deployment_type.
+        """
+        return self.arguments.get('bootstrapless_config', {})
+
+    def handle(self) -> str:
+        """
+        Main Sceptre CDK Handler function
+
+        Returns:
+            str - The CDK synthesised CloudFormation template
+        """
+        stack_class: Type[SceptreCdkStack] = self._importer.import_class(self.cdk_template_path, self.cdk_class_name)
+        if self.deployment_type == 'bootstrapped':
+            context, builder = self._get_bootstrapped_builder()
+        elif self.deployment_type == "bootstrapless":
+            builder = self._get_bootstrapless_builder()
+            context = self.cdk_context
+        else:
+            # It shouldn't be possible to get here due to the json schema validation
+            raise ValueError("deployment_type must be 'bootstrapped' or 'bootstrapless'")
+
+        template_dict = builder.build_template(
+            stack_class,
+            context,
+            self.sceptre_user_data
+        )
+        return yaml.safe_dump(template_dict)
+
+    def _get_bootstrapped_builder(self) -> Tuple[Optional[dict], BootstrappedCdkBuilder]:
+        builder = self._bootstrapped_cdk_builder_class(self.logger, self.connection_manager)
+        # The qualifier might already be in the context, in which case we don't need to do
+        # anything
+        if self.cdk_context and QUALIFIER_CONTEXT_KEY in self.cdk_context:
+            return self.cdk_context, builder
+        # As a convenience, the qualifier can be set as its own argument to simplify the
+        # configuration. If it's passed this way, we need to add it to whatever context dict there
+        # is, if one exists; We'll make one if it doesn't.
+        if self.bootstrap_qualifier:
+            context = self.cdk_context or {}
+            context[QUALIFIER_CONTEXT_KEY] = self.bootstrap_qualifier
+            return context, builder
+        # If there's no qualifier specified anywhere, we're falling back to either the specified
+        # context or the default CDK context (if no context is specified on the handler)
+        return self.cdk_context, builder
+
+    def _get_bootstrapless_builder(self) -> BootstraplessCdkBuilder:
+        return self._bootstrapless_cdk_builder_class(
+            self.logger,
+            self.connection_manager,
+            self.bootstrapless_config
+        )
+
+    def validate(self):
+        super().validate()
+        self._check_prerequisites()
+
+    def _check_prerequisites(self) -> None:
+        """
+        Checks the command and Node package requirements for the handler.
+
+        Raises:
+            SceptreException: Command prerequisite not found
+            SceptreException: Node Package prerequisite not found
+        """
+        # Check Command Prerequisites
+        cmd_prerequisites = [
+            'node',
+            'npx'
```

Comment:
should we check for `docker` command in case deployment to ECR is needed?

Reply:
Good question. We would only actually require the use of docker IF there are image assets to publish. If there aren't image assets, we don't need docker. But that's a good call-out. I think I'll add that to the docs but not enforce it here.